src/sentry/tasks/releasemonitor.py

```diff
@@ -169,28 +165,43 @@ def adopt_releases(org_id, totals):
                 total_releases = len(environment_totals["releases"])
                 for release in environment_totals["releases"]:
                     threshold = 0.1 / total_releases
-                    if environment_totals["total_sessions"] != 0 and (
-                        environment_totals["releases"][release]
+                    if (
+                        environment != ""
+                        and environment_totals["total_sessions"] != 0
+                        and environment_totals["releases"][release]
                         / environment_totals["total_sessions"]
                         >= threshold
                     ):
                         try:
                             rpe = ReleaseProjectEnvironment.objects.get(
                                 project_id=project_id,
-                                release_id__in=Release.objects.filter(
+                                release_id=Release.objects.get(
                                     organization=org_id, version=release
-                                ).values("id")[:1],
+                                ).id,
                                 environment__name=environment,
                                 environment__organization_id=org_id,
                             )
                             adopted_ids.append(rpe.id)
                             if rpe.adopted is None:
                                 rpe.update(adopted=timezone.now())
-                        except ReleaseProjectEnvironment.DoesNotExist as exc:
-                            metrics.incr(
-                                "sentry.tasks.process_projects_with_sessions.skipped_update"
-                            )
-                            capture_exception(exc)
+                        except ReleaseProjectEnvironment.DoesNotExist:
+                            metrics.incr("sentry.tasks.process_projects_with_sessions.creating_rpe")
+                            try:
+                                env = Environment.objects.get(
+                                    name=environment, organization_id=org_id
+                                )
+                                rel = Release.objects.get(organization=org_id, version=release)
+                                ReleaseProjectEnvironment.objects.create(
+                                    project_id=project_id,
+                                    release_id=rel.id,
+                                    environment=env,
+                                    adopted=timezone.now(),
+                                )
```

Comment:
Hmm, since we're doing this should we also make sure that the `ReleaseProject` and `ReleaseEnvironment` exist?

Reply:
Hmm yeah I can do that for now. I won't create them but I'll make sure they exist before creating `ReleaseProjectEnvironment`. I'll watch this and see how many cases this covers (hopefully the same amount).

Reply:
Tbh I wasn't sure how you'd feel about the extra queries, but I guess it's okay given the context (rarely happens, and what other option exists?)

Reply:
I kind of wrote this ambiguously, but I actually meant we should probably create them. Not sure if things in our system will end up in a weird state if only some of these rows exist.